Updates for use in disconnected environments

bagnaram opened the pull request:
Fixes git repository URLS
Remove dependency on EPEL (for use in disconnected installs)
Use image-inspector base image instead of installing it as a package

Commit:
Remove image-inspector RPM from base and use the docker image instead.

remove epel from base image dependency

JQ binary installation

fix wget to curl in dockerfile

add redirect for jq install

Parameterize the OCP release version in the Dockerfiles
Rename the git repository to point to redhat-cop
Additional tweaks to make working in disconnected environments

added a hack to get latest version of the image-inspector

change ARG to ENV since we arent really referencing images here

replace the git repository names to point to redhat-cop

Added rhel7 back into imagestreams
Removed --openscap-html-report to address https://github.com/openshift/image-inspector/issues/105
Fixed bug in imagestream instead of direct Docker image

fixed the typo update scan message in the operator

modify group_vars to parse imagestreams as template instead of single image

Commit:
Remove image-inspector RPM from base and use the docker image instead.

Commit:
remove epel from base image dependency

Commit:
JQ binary installation

Commit:
fix wget to curl in dockerfile

Commit:
add redirect for jq install

Commit:
added a hack to get latest version of the image-inspector

Commit:
parameterized OCP_VERSION

Commit:
change ARG to ENV since we arent really referencing images here

Commit:
comment the arg and try removing it

Commit:
type

Commit:
test

Commit:
test

Commit:
replace the git repository names to point to redhat-cop

Commit:
Parameterize the OCP release version in the Dockerfiles
Rename the git repository to point to redhat-cop
Additional tweaks to make working in disconnected environments

Commit:
Merge branch 'master' of https://github.com/bagnaram/image-scanning-signing-service

sabre1041 commented on files/imagestreams/images.yml:
```diff
@@ -1,6 +1,6 @@
 apiVersion: v1
```
The rhel7 ImageStream is still required as it is utilized by the s2i golang image. I would recommend turning this object into a list so that both imagestreams can be utilized

bagnaram replied:
NP. I made it into a template like the others

sabre1041 commented on files/builds/image-sign-scan-base/template.yml:
```diff
@@ -50,19 +50,19 @@ parameters:
   name: FROM_IMAGE
   required: true
-  value: rhel7:7.5
+  value: openshift3/image-inspector:latest
```
This should just be `image-inspector:latest` as it refers to an ImageStreamTag

sabre1041 reviewed (changes requested):
@bagnaram This looks really good! There are a few issues that came up during testing. Once corrected, we can integrate

bagnaram commented:
Thanks! I will go ahead and test then squash

Commit:
Added rhel7 back into imagestreams
Removed --openscap-html-report to address https://github.com/openshift/image-inspector/issues/105
Fixed bug in imagestream instead of direct Docker image

Commit:
fixed the typo update message

Commit:
add missing params file

Commit:
remove verbose from the scan-image script

sabre1041 commented on files/imagestreams/images.yml:
```diff
@@ -1,6 +1,21 @@
 apiVersion: v1
-kind: ImageStream
+kind: Template
```
With the change to a template, the applier inventory must also be updated to process the template. Otherwise it just adds it to the project. The following change to the `all.yml` should handle it

```
diff --git a/inventory/group_vars/all.yml b/inventory/group_vars/all.yml
index 6bdd7e1..df67f71 100644
--- a/inventory/group_vars/all.yml
+++ b/inventory/group_vars/all.yml
@@ -50,8 +50,9 @@
             - core
     - object: imagestreams
       content:
-        - name: rhel
-          file: "{{ inventory_dir }}/../files/imagestreams/images.yml"
+        - name: images
+          template: "{{ inventory_dir }}/../files/imagestreams/images.yml"
+          params: "{{ inventory_dir }}/../files/imagestreams/params"
           namespace: "image-management"
           tags:
             - core
```

bagnaram replied:
Thanks for assisting @sabre1041 . Fixed the group_vars

sabre1041 commented on images/image-sign-scan-base/bin/scan-image:
```diff
@@ -15,4 +15,4 @@ DOCKERCFG_SECRET_NAME=$(echo "${SA_JSON}" | jq -r ".imagePullSecrets[] | select(
 
-image-inspector -chroot --path=${IMAGE_CONTENT} --scan-results-dir=${RESULTS} --scan-type=openscap --openscap-html-report --dockercfg=$DOCKER_CFG_HOME/.dockercfg --image=$IMAGE -serve=0.0.0.0:8080
\ No newline at end of file
+image-inspector -chroot --path=${IMAGE_CONTENT} --scan-results-dir=${RESULTS} --scan-type=openscap  --dockercfg=$DOCKER_CFG_HOME/.dockercfg --image=$IMAGE -serve=0.0.0.0:8080
```
The html report should still be generated to support the CI/CD integration.

bagnaram replied:
There is actually a bug in image-inspector when generating HTML reports. See https://github.com/openshift/image-inspector/issues/105
Not sure how to address this until the PR gets merged besides taking the HTML generation out.

sabre1041 replied:
This potentially might have been why I initially went down the custom image route with the RPM instead of the OCP image.

I am on the fence on which to move forward with now because the lack of html reports is a big loss in the value that is provided by this solution that can be shown to management

bagnaram replied:
I agree, it would be a shame to lose. I'l compare the RPM distributed version of image-inspector with the container-provided one.

bagnaram replied:
The image tested in the PR is includes `image-inspector-2.4.0-3.el7.x86_64`

bagnaram replied:
The image tested in the PR is `openshift3/image-inspector:v3.11.51-3` includes `image-inspector-2.4.0-3.el7.x86_64`
The previous image-sign-scan-base installs the latest copy from the `rhel-7-server-ose-3.11-rpms`  or whatever version of OCP being used. (in the previous case v3.9) The package version for 3.9 is `image-inspector-2.1.2-2.el7.x86_64` which doesn't suffer from the above bug. However it is missing crucial header handling which breaks the retrieval of the CVE data since a recent update on the RH servers hosting it. Either way, it doesn't appear to work.

sabre1041 replied:
Fun! I just went ahead and tagged the repository as it stands today to `v1.0.0` so we have a way to utilize somewhat working functionality. Would you be able to document what functionality will be lost due to the associated bugs? Otherwise, we should be good to move forward.

bagnaram replied:
I added an additional Known issues section

sabre1041 reviewed (changes requested):
@bagnaram Ran through the latest changes. A few more items to address. Are you planning on performing a squash before  the changes are integrated?

Commit:
Merge branch 'master' of https://github.com/bagnaram/image-scanning-signing-service

bagnaram commented:
Squashed

Commit:
added galaxy to gitignure
fixed URL in Jenkinsfile

Commit:
added Known Issues section to README

sabre1041 reviewed (approved):
This looks great. Thanks @bagnaram
